tedana/metrics/collect.py

```diff
@@ -0,0 +1,140 @@
+    algorithm : {'kundu_v2', 'kundu_v3', None}, optional
+        Decision tree to be applied to metrics. Determines which maps will be
+        generated and stored in seldict. Default: None
+    label : :obj:`str` or None, optional
+        Prefix to apply to generated files. Default is None.
+    out_dir : :obj:`str`, optional
+        Output directory for generated files. Default is current working
+        directory.
+    verbose : :obj:`bool`, optional
+        Whether or not to generate additional files. Default is False.
+
+    Returns
+    -------
+    comptable : (C x X) :obj:`pandas.DataFrame`
+        Component metric table. One row for each component, with a column for
+        each metric. The index is the component number.
+    seldict : :obj:`dict` or None
+        Dictionary containing component-specific metric maps to be used for
+        component selection. If `algorithm` is None, then seldict will be None as
+        well.
+    betas : :obj:`numpy.ndarray`
+    mmix_new : :obj:`numpy.ndarray`
+    """
+    RepLGR.info('The following metrics were calculated: {}.'.format(', '.join(metrics)))
+
+    if not (data_cat.shape[0] == data_optcom.shape[0] == mask.sum()):
+        raise ValueError('First dimensions (number of samples) of data_cat ({0}), '
+                         'data_optcom ({1}), and mask ({2}) do not '
+                         'match'.format(data_cat.shape[0], data_optcom.shape[0],
+                                        mask.shape[0]))
+    elif data_cat.shape[1] != len(tes):
+        raise ValueError('Second dimension of data_cat ({0}) does not match '
+                         'number of echoes provided (tes; '
+                         '{1})'.format(data_cat.shape[1], len(tes)))
+    elif not (data_cat.shape[2] == data_optcom.shape[1] == mixing.shape[0]):
+        raise ValueError('Number of volumes in data_cat ({0}), '
+                         'data_optcom ({1}), and mixing ({2}) do not '
+                         'match.'.format(data_cat.shape[2], data_optcom.shape[1], mixing.shape[0]))
+
+    mixing = mixing.copy()
+    comptable = pd.DataFrame(index=np.arange(n_components, dtype=int))
+
+    # Metric maps
+    weights = calculate_weights(data_optcom, mixing_z)
+    signs = determine_signs(weights, axis=0)
+    weights, mixing = flip_components(weights, mixing, signs=signs)
+    optcom_betas = calculate_betas(data_optcom, mixing)
+    PSC = calculate_psc(data_optcom, optcom_betas)
+
+    # compute betas and means over TEs for TE-dependence analysis
+    Z_maps = calculate_z_maps(weights)
+    F_T2_maps, F_S0_maps = calculate_f_maps(mixing, data_cat, tes, Z_maps)
+
+    (Z_clmaps, F_T2_clmaps, F_S0_clmaps,
+     Br_T2_clmaps, Br_S0_clmaps) = spatial_cluster(
+        F_T2_maps, F_S0_maps, Z_maps, optcom_betas, mask, n_echos)
+
+    # Dependence metrics
+    if any([v in metrics for v in ['kappa', 'rho']]):
+        comptable = calculate_dependence_metrics(comptable, F_T2_maps, F_S0_maps, Z_maps)
```

Comment:
Based on our discussions in-person, it sounds like we're thinking maybe we should aim for these functions to return arrays (_not_ `comptable`), so this function should be updated as with the ones below.

Reply:
Done.